common/src/main/java/com/tc/net/protocol/transport/ClientConnectionEstablisher.java

```diff
@@ -294,7 +294,7 @@ void handleConnectException(Exception e, boolean logFullException, Logger logger
       try {
         Thread.sleep(CONNECT_RETRY_INTERVAL);
       } catch (InterruptedException e1) {
-        Thread.currentThread().interrupt();
+        LOGGER.info("Async reconnect thread woken by interrupt " + transport);
       }
```

Comment:
This seems problematic if a connection is closed while trying to connect or reconnect.  I'd like to understand why the connect loop is not breaking out of the loop if interrupted.  The only reason this thread should be interrupted is if the connection is closed.

Reply:
Right... but then what's going on with the interrupt that happens when `putConnectionRequest(...)` is called?